### ISSUE TITLE
74194 shift direction implemented wrong

### ISSUE THREAD

**Issue by altelch** [hneemann/Digital#921]
Hi,

according to the datasheet https://cdn-reichelt.de/documents/datenblatt/A240/SN74194_74LS194%23TEX.pdf S0=H, S1=L the register should shift right from Pin 15 (A) to 12 (D). Your version shifts left. Direction is exchanged to what is specified in the datasheet.

Greetings,

          Altelch.

**Comment by coreidcc** [comment]
that fix works as well but is rather confusing. have a signal S0 on bit 1 and S1 on bit 0 of the selector.

**Comment by altelch** [comment]
👍

**Comment by tcwan** [comment]
I encountered this issue recently when trying to use it to implement the Ring Counter example (from "Fig. 8-42, J. F. Wakerly, Digital Design: Principles and Practices, 4th Edition, Pearson, 2014")

Actually the issue is not that the logic was implemented incorrectly in the previous verison (release v.029, 2-12-2022). The issue was that the numbering of the pins Q0-Q3 was in the opposite order than that defined in the datasheet. In the datasheet, QA (pin 15) == Q3 (pin 12), while QD (pin 12) == Q0 (pin 15). Similarly for inputs A-D. If you rearrange the pin ordering for the 74194 part in the simulator to follow the pin ordering for the datasheet, it will work correctly. 

TLDR: The pin labels in the simulator are correct, it is the pinouts that are wrongly placed for D0-D3 and Q0-Q3.
